hbase-zookeeper/src/main/java/org/apache/hadoop/hbase/zookeeper/ZKUtil.java

```diff
@@ -741,6 +741,10 @@ public static int getNumberOfChildren(ZKWatcher zkw, String znode)
     if (nodes != null) {
       List<NodeAndData> newNodes = new ArrayList<>();
       for (String node : nodes) {
+        if (Thread.interrupted()) {
+          // Partial data should not be processed. Cancel processing by sending empty list.
+          return new ArrayList<>();
```

Comment:
returning `Collections.emptyList()` would be better

Reply:
done